src/js/controller/controller.js

```diff
@@ -577,7 +592,9 @@ Object.assign(Controller.prototype, {
             _actionOnAttach = null;
             checkAutoStartCancelable.cancel();
 
-            _model.set('pauseReason', _getReason(meta));
+            const pauseReason = _getReason(meta);
+            _model.set('pauseReason', pauseReason);
+            _model.set('playOnViewable', (pauseReason === 'viewability'));
```

Comment:
So the only reason you added "playOnViewable" and "pauseOnViewable" model properties was to access them in instream? Instream has access to the controller. You should not use the model for things that (1)  you don't need to share throught the application (2) you can access as properties from other objects (3) you don't need "change" listeners for.

Reply:
`playOnViewable` already existed on the model before these changes. I originally tried just using `playOnViewable` to implement the autoPause functionality but it would not work properly and involved a lot more logic than would be needed by just adding another property on the model.

I'm also setting it on the model because I believe we do need to share this throughout other components. Please see my replies to your feedback in the program-controller and instream-adapter files.

Reply:
instream has access to the controller, so why not `controller.on/off('viewable', controller.play)`?